### PR TITLE
feat(strm-1259): remove billing id usage

### PR DESCRIPTION
**Pull request by bobvandenhoogen**
BREAKING CHANGE: authentication has changed

### CODE REVIEW

File: lib/src/main/java/io/strmprivacy/driver/client/AuthService.java
Comment (trietsch) [inline lib/src/main/java/io/strmprivacy/driver/client/AuthService.java]: You can use the `HttpHeader` constants.

File: lib/src/main/java/io/strmprivacy/driver/domain/Config.java
Comment (trietsch) [inline lib/src/main/java/io/strmprivacy/driver/domain/Config.java]: For consistency sake, please split the URL parts, as before.

File: lib/src/test/java/io/strmprivacy/driver/client/StrmPrivacyClientTest.java
Comment (trietsch) [inline lib/src/test/java/io/strmprivacy/driver/client/StrmPrivacyClientTest.java]: Did you actually update the test itself? You should create the specific mocks with Wiremock in order to properly test it.

**Review by trietsch — CHANGES_REQUESTED** [review]

File: lib/src/main/java/io/strmprivacy/driver/client/AuthProvider.java
Comment (trietsch) [inline lib/src/main/java/io/strmprivacy/driver/client/AuthProvider.java]: Use pascal case.

File: lib/src/main/java/io/strmprivacy/driver/client/AuthProvider.java
Comment (trietsch) [inline lib/src/main/java/io/strmprivacy/driver/client/AuthProvider.java]: casing?

File: lib/src/main/java/io/strmprivacy/driver/domain/Config.java
Comment (trietsch) [inline lib/src/main/java/io/strmprivacy/driver/domain/Config.java]: Convention is to start paths on a host with `/`.

File: lib/src/main/java/io/strmprivacy/driver/client/AuthService.java
Comment (trietsch) [inline lib/src/main/java/io/strmprivacy/driver/client/AuthService.java]: The slash should be removed here for the path. That should be included in the path parameter.

**Review by trietsch — CHANGES_REQUESTED** [review]

**Review by trietsch — CHANGES_REQUESTED** [review]
~Once we can merge this, **this PR should be merged as breaking change**~ -> no changes required.

**Review by trietsch — APPROVED** [review]
Approved, but once we can merge this, **this PR should be merged as breaking change**

**Review by ivan-p92 — APPROVED** [review]

**Pull request merged**
